tests/durability/test_disk_failure_recovery.py

```diff
@@ -338,16 +401,108 @@ def test_sns_repair_fail_disk_equal_to_k(self):
         LOGGER.info("Step 10: Read data written in step 1")
         resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
                                                     log_prefix=self.test_prefix[0],
-                                                    skipwrite=True,
+                                                    skipwrite=True, skipcleanup=True)
+        assert_utils.assert_true(resp[0], resp[1])
+
+        LOGGER.info("Step 11: Do IOs(Write and Read)")
+        self.test_prefix.append('test-36393-after-recovery')
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix[-1],
+                                                    skipcleanup=True)
+        assert_utils.assert_true(resp[0], resp[1])
+
+        LOGGER.info("COMPLETED: Test SNS repair works fine with failed disks "
+                    "are less than K(parity units) and from different cvg")
+
+    @pytest.mark.lc
+    @pytest.mark.tags("TEST-36394")
+    def test_sns_repair_fail_disk_diff_cvg_equal_to_k(self):
+        """
+        Validate SNS repair works fine with failed disks are equal to K(parity units)
+        and from different cvg
+        """
+        LOGGER.info("STARTED: Validate SNS repair works fine with failed disks "
+                    "are equal to K(parity units) and from different cvg")
+
+        LOGGER.info("Step 1: Do IOs(Write and Read)")
+        users = self.mgnt_ops.create_account_users(nusers=1)
+        self.test_prefix.append('test-36394')
+        self.s3_clean = users
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix[-1],
+                                                    skipcleanup=True)
+        assert_utils.assert_true(resp[0], resp[1])
+
+        LOGGER.info("Step 2: Get degraded byte count before failing the disk")
```

Comment:
can we add this to setup_method as I can see we are taking this count before every test

Reply:
No, as we are doing IOs before this step so this count might change after step 1.